### PR TITLE
Add transforms

### PR DESCRIPTION
**Pull request by LielinJiang**
1. add transforms
2. add unit test for transform and dataset
3. refine models api
4. add some test data
5. update image classification readme

### CODE REVIEW

File: models/mobilenetv1.py [hunk 1]
```diff
@@ -227,28 +235,34 @@ def __init__(self, scale=1.0, class_dim=1000):
...
+            self.pool2d_avg = Pool2D(pool_type='avg', global_pooling=True)
 
-        self.out = Linear(
-            int(1024 * scale),
-            class_dim,
-            act='softmax',
-            param_attr=ParamAttr(
-                initializer=MSRA(), name=self.full_name() + "fc7_weights"),
-            bias_attr=ParamAttr(name="fc7_offset"))
+        if num_classes > -1:
```
Comment (heavengate) [inline models/mobilenetv1.py:241]: 这里判断统一下吧，我看其他文件里是`if num_classes > 0`

File: tests/test_transforms.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+import sys
+sys.path.append('..')
```
Comment (heavengate) [inline tests/test_transforms.py:17]: 不建议这样设置目录，感觉在README里说明在根目录`export PYTHONPATH=`pwd`:$PYTHONPATH`比较好

Comment (LielinJiang) [reply tests/test_transforms.py]: done, thanks. beause it is a unittest file, i add this comment in the head.

File: models/mobilenetv1.py [hunk 3]
```diff
@@ -111,13 +111,21 @@ class MobileNetV1(Model):
 
     Args:
         scale (float): scale of channels in each layer. Default: 1.0.
-        class_dim (int): output dim of last fc layer. Default: 1000.
+        num_classes (int): output dim of last fc layer. Default: -1.
+        with_pool (bool): use pool or not. Default: False.
```
Comment (qingqing01) [inline models/mobilenetv1.py:115]: Need to specify  where pool is used.

File: models/mobilenetv1.py (hunk 1 above)
Comment (LielinJiang) [reply models/mobilenetv1.py:241]: done, thanks.

File: models/mobilenetv1.py [hunk 4]
```diff
@@ -111,13 +111,21 @@ class MobileNetV1(Model):
 
     Args:
         scale (float): scale of channels in each layer. Default: 1.0.
-        class_dim (int): output dim of last fc layer. Default: 1000.
+        num_classes (int): output dim of last fc layer. Default: -1.
```
Comment (qingqing01) [inline models/mobilenetv1.py:114]: need more clearly.   -1 means not  include the fully-connected layer at the top/output of the network.

File: models/resnet.py [hunk 5]
```diff
@@ -164,11 +164,21 @@ class ResNet(Model):
...
+        with_pool (bool): use pool or not. Default: False.
+        classifier_activation (str): activation for the last fc layer. Default: 'softmax'.
     """
 
-    def __init__(self, Block, depth=50, num_classes=1000):
+    def __init__(self,
+                 Block,
+                 depth=50,
+                 num_classes=-1,
+                 with_pool=False,
```
Comment (qingqing01) [inline models/resnet.py:175]: Why default value is -1 and False?  Whether we keep the defalut value for ImageNet?

File: transform/transforms.py [hunk 6]
```diff
@@ -0,0 +1,467 @@
...
+
+    Args:
+        mode: Output mode of input. Use "CHW" mode by default.
+    """
+
+    def __init__(self, mode="CHW"):
+        assert mode in ["CHW"], "unsupported mode: {}".format(mode)
+        self.mode = mode
+
+    def __call__(self, img):
```
Comment (qingqing01) [inline transform/transforms.py:302]: 上面一些class加了注释，一些没加。 这块加下，输入的img的mode和格式

File: models/mobilenetv1.py (hunk 3 above)
Comment (LielinJiang) [reply models/mobilenetv1.py]: done, thanks.

File: models/mobilenetv1.py (hunk 4 above)
Comment (LielinJiang) [reply models/mobilenetv1.py]: done, thanks.

File: models/resnet.py (hunk 5 above)
Comment (LielinJiang) [reply models/resnet.py]: done, thanks.

File: transform/transforms.py (hunk 6 above)
Comment (LielinJiang) [reply transform/transforms.py:305]: done, thanks.

**Pull request merged**